lamindb_setup/_set_managed_storage.py

```diff
@@ -28,14 +29,22 @@ def set_managed_storage(root: UPathStr, **fs_kwargs):
         raise ValueError(
             "Can't add additional managed storage locations for instances that aren't managed through the hub."
         )
-    ssettings = init_storage(
+
+    ssettings, hub_record_status = init_storage(
         root=root, instance_id=settings.instance._id, register_hub=True
     )
     if ssettings._instance_id is None:
         raise ValueError(
             f"Cannot manage storage without write access: {ssettings.root}"
         )
+
+    try:
+        register_storage_in_instance(ssettings)
+    except Exception as e:
+        if hub_record_status == "hub_record_created" and ssettings._uuid is not None:
```

Comment:
Also this deserves extensive commenting.

Reply:
I will comment.